edb/server/protocol/ai_ext.py

```diff
@@ -1602,28 +1596,18 @@ async def _handle_rag_request(
 
         role = message["participant_role"].lower()
 
-        try:
-            by_role = messages[role]
-        except KeyError:
-            by_role = messages[role] = []
-
-        by_role.append((dict(role=role, content=content)))
-
-    for role, role_messages in custom_prompt_messages.items():
-        try:
-            by_role = messages[role]
-        except KeyError:
-            by_role = messages[role] = []
-
-        by_role.extend(role_messages)
+        prompt_messages.append(dict(role=role, content=content))
 
+    messages = [prompt_messages[0]]
+    messages.extend(custom_prompt_messages)
+    messages.append(prompt_messages[1])
```

Comment:
hmm this is good question... I used default prompt but actually there can be 1 or more messages too I think

Reply:
@elvis I left this same message in the Engineering:
I’m a bit confused with how we are supposed to use EdgeDB AI ext.
Why can you add multiple messages to the prompt? Why it doesn’t only have the system message?
From what I tested I see that prompt besides system message needs to have user message that includes user question.
These 2 messages are then sent to the AI provider.
User question is only sent to the AI provider if it is included in the prompt -> user_message like on the screenshot.
User question is not seen like a complete standalone message that is included in the messages array sent to the AI provider.

Maybe I missunderstood how our ext should work. Because to the AI providers (for sure OpenAI) the messages array should have this order:
system message,
chat history,
last user question that needs answer.